src/utils/actions.ts

```diff
@@ -13,8 +14,11 @@ const getView = (headers: { origin?: string | string[] }, hosts: Hostname[]): st
   VIEW_UNKNOWN_NAME;
 
 const getBaseAction = (request: FastifyRequest, hosts: Hostname[]): Partial<BaseAction> => {
-  const { member, ip, headers } = request;
+  const { member, headers } = request;
   const view = getView(headers, hosts);
+  // warning: addresses might contained spoofed ips
+  const addresses = forwarded(request.raw);
+  const ip = addresses.pop();
```

Comment:
is this guaranteed to return a defined value?

Reply:
It always returns at least `[remoteAddress]` so yep!